### PR TITLE
Update documents

### LINKED ISSUE #242: Architecture sequence diagram uses deletePerson instead of deleteStudent

### LINKED ISSUE #243: Storage class diagram has JSonAdaptedPerson and jsonAdaptedTag
Should remove jsonAdaptedTag and change jsonAdaptedPerson to jsonAdaptedStudent

### PR DESCRIPTION
**Pull request by Feng1231**
Changes:
- Add add command design consideration
- Change person instance to student in puml
- Remove tag related code in puml
- Delete undo diagrams
- Delete invalid tutorial links. This is because the tutorial links is for AB3 not UGTeach
- Update format for use case extensions. The extensions now look like this:
image
- Update all instance of ordered lists to use `1.` only
- Removed this portion, as this portion is pertaining to better implementation for tag
image
- Combine similar extensions in usecase to one

closes #242 
closes #243

### CODE REVIEW

File: docs/diagrams/StorageClassDiagram.puml [hunk 1]
```diff
@@ -37,7 +37,7 @@ Storage -right-|> AddressBookStorage
 JsonUserPrefsStorage .up.|> UserPrefsStorage
 JsonAddressBookStorage .up.|> AddressBookStorage
 JsonAddressBookStorage ..> JsonSerializableAddressBook
-JsonSerializableAddressBook --> "*" JsonAdaptedPerson
-JsonAdaptedPerson --> "*" JsonAdaptedTag
+JsonSerializableAddressBook --> "*" JsonAdaptedStudent
+JsonAdaptedStudent --> "*" JsonAdaptedTag
```
Comment (Soumyadip-cmd) [inline docs/diagrams/StorageClassDiagram.puml:41]: Can perhaps remove JsonAdaptedTag

**Review by Soumyadip-cmd — COMMENTED** [review]
Can perhaps link to issues #242 and #243. Can help to fix the architecture sequence diagram too. Thanks in advance.

File: docs/diagrams/StorageClassDiagram.puml (hunk 1 above)
Comment (Feng1231) [reply docs/diagrams/StorageClassDiagram.puml:41]: Ok

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -200,18 +192,17 @@ Similarly, how an AddCommand operation goes through the `Model` component is sho
...
+  * Cons: Command might get too long.
 
-* **Alternative 2:** Individual command knows how to undo/redo by
-  itself.
-  * Pros: Will use less memory (e.g. for `delete`, just save the student being deleted).
-  * Cons: We must ensure that the implementation of each individual command are correct.
+* **Alternative 2:** Key in details for students in multiple steps.
+  * Pros: Step by step adding of details might be more convenient for new users.
+  * Cons: Hard to implement, especially with optional fields and compulsory fields. It is also
+  not user-friendly for users who type fast due to multi line commands required to add a student.
```
Comment (teamotitan) [inline docs/DeveloperGuide.md:204]: A consideration for you(I might be wrong so you can ignore if its not suitable)
Alternative 1
"Key in the details of a student in a single command"
"Easy to implement, as only one...."

Alternative 2
"Key in the details of a student in multiple steps"
"A step by step way of adding details..."
"type fast, as multi-line commands are required..."

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -299,73 +290,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 1b1. System shows an empty list.<br> 
   Use case ends.
 
 **Use case: UC03 - Read total earnings**
 
 **MSS**
 
-1. User enters command to read total earnings and total money owed.
-2. System displays total earnings and total money owed to the user.
+1. User enters command to read total earnings and total money owed.
```
Comment (teamotitan) [inline docs/DeveloperGuide.md:335]: maybe amount instead of money?

File: docs/diagrams/DeleteSequenceDiagram.puml [hunk 4]
```diff
@@ -49,7 +49,7 @@ deactivate AddressBookParser
 LogicManager -> DeleteCommand : execute(m)
 activate DeleteCommand
 
-DeleteCommand -> Model : deletePerson(1)
+DeleteCommand -> Model : deleteStudent(1)
```
Comment (teamotitan) [inline docs/diagrams/DeleteSequenceDiagram.puml:52]: Great catch!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (Feng1231) [reply docs/DeveloperGuide.md:335]: Noted

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Feng1231) [reply docs/DeveloperGuide.md]: I have rephrased this part. Please take a look.

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -299,73 +290,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 2a2. User enters command with new index.<br>
+  Steps 2a1-2a2 are repeated until index entered is correct.<br>
+  Use case resumes from step 3.<br><br>
+
+
+* 2b. System detects error in format of entered command.
+    * 2b1. System displays error message and does not clear command line.
+    * 2b2. User enters new command.<br<
+  Steps 2b1-2b2 are repeated until all details entered are correct.<br>
+  Use case resumes from step 3.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:371]: The 2 extensions here seem almost the same. Can maybe join to make just 1 extension or can specify what are the errors in the 2 extensions.

File: docs/SettingUp.md [hunk 6]
```diff
@@ -52,11 +52,5 @@ If you plan to use Intellij IDEA (highly recommended):
...
 
-   When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [AddressBook’s architecture](DeveloperGuide.md#architecture).
+   When you are ready to start coding, we recommend that you get some sense of the overall design by reading about [UGTeach’s architecture](DeveloperGuide.md#architecture).
 
-1. **Do the tutorials**
-   These tutorials will help you get acquainted with the codebase.
-
-   * [Tracing code](tutorials/TracingCode.md)
-   * [Adding a new command](tutorials/AddRemark.md)
-   * [Removing fields](tutorials/RemovingFields.md)
```
Comment (Soumyadip-cmd) [inline docs/SettingUp.md:62]: Not sure if these really need to be deleted.

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -299,73 +290,85 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
-* Use case resumes from step 2.
+    * 1a2. User enters new command.<br>
+  Steps 1a1-1a2 are repeated until all details entered are correct.<br>
+  Use case resumes from step 2.<br><br>
+  
+* 1b. System detects error in parameters.
+    * 1b1. System displays error message and does not clear command line.
+    * 1b2. User enters command with correct parameters.<br>
+  Steps 1b1-1b2 are repeated until all details entered are correct.<br>
+  Use case resumes from step 2.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:309]: Should we follow this standard or the standard in the use cases for pay and settle? This one has extensions for error in entered command and entered parameters as separate but in pay and settle they are considered in the same use case

**Review by Soumyadip-cmd — COMMENTED** [review]
Just some considerations

File: docs/SettingUp.md (hunk 6 above)
Comment (Feng1231) [reply docs/SettingUp.md:62]: [If you click into these, it actually throws an error, even in the AB3 DG. see here](https://se-education.org/addressbook-level3/SettingUp.html)

File: docs/DeveloperGuide.md (hunk 7 above)
Comment (Feng1231) [reply docs/DeveloperGuide.md:309]: Good question

File: docs/DeveloperGuide.md (hunk 5 above)
Comment (Feng1231) [reply docs/DeveloperGuide.md:371]: Thanks for catching this.

File: docs/SettingUp.md (hunk 6 above)
Comment (Soumyadip-cmd) [reply docs/SettingUp.md:62]: Alright then makes sense to delete. Thanks for catching it.

File: docs/DeveloperGuide.md [hunk 8]
```diff
@@ -315,73 +307,72 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
 **Extensions**
 
 * 2a. System detects error in format of entered command.
     * 2a1. System displays error message and does not clear command line.
-    * 2a2. User enters new command.
-* Steps 2a1-2a2 are repeated until all details entered are correct.
-* Use case resumes from step 3.
+    * 2a2. User enters command with new index.<br>
+  Steps 2a1-2a2 are repeated until index entered is correct.<br>
+  Use case resumes from step 3.
```
Comment (Soumyadip-cmd) [inline docs/DeveloperGuide.md:375]: Can perhaps consider making the use case like UC06 and 07

**Review by Soumyadip-cmd — COMMENTED** [review]
Just some consideration. Otherwise looks good.

**Review by Soumyadip-cmd — APPROVED** [review]
LGTM

**Pull request merged**